snapcraft/tests/fixture_setup.py

```diff
@@ -29,16 +33,94 @@ def setUp(self):
         os.chdir(self.path)
 
 
-class StagingStore(fixtures.Fixture):
+class _FakeServerRunning(fixtures.Fixture):
+
+    # To be defined by child fixtures.
+    fake_server = None
+
+    def setUp(self):
+        super().setUp()
+        self._start_fake_server()
+
+    def _start_fake_server(self):
+        server_address = ('', 0)
+        self.server = self.fake_server(server_address)
+        server_thread = threading.Thread(target=self.server.serve_forever)
+        server_thread.start()
+        self.addCleanup(self._stop_fake_server, server_thread)
```

Comment:
This is racy, there is no guarantee that the server will be listening to the socket before the client emit the first request.